Public/Src/Engine/UnitTests/Scheduler/PipFingerprinterTests.cs

```diff
@@ -400,6 +411,91 @@ public void TestUserProfileStaysOutOfFingerprints()
             XAssert.IsTrue(json.Contains(Path.GetFileName(Environment.GetFolderPath(Environment.SpecialFolder.InternetCache)).ToUpperInvariant()));
         }
 
+        [Fact]
+        public void TestModifiedExecutableWhenUntrackedStaysOutOfFingerprints()
+        {
+            var exectuablePath = "/x/pkgs/tool.exe";
+            var pathTable = m_context.PathTable;
+            var executable = FileArtifact.CreateSourceFile(AbsolutePath.Create(pathTable, X(exectuablePath)));
+
+            var dependencies = new HashSet<FileArtifact>() { executable };
+            var process = GetDefaultProcessBuilder(pathTable, executable)
+                .WithDependencies(dependencies)
+                .WithUntrackedPaths(new AbsolutePath[] { AbsolutePath.Create(pathTable, X(exectuablePath)) })
+                .Build();
+
+            MountPathExpander expander = new MountPathExpander(pathTable);
+
+            var fingerprinter1 = new PipContentFingerprinter(
+                m_context.PathTable,
+                GetDummyContentHash(5),
+                ExtraFingerprintSalts.Default(),
+                pathExpander: expander)
+            {
+                FingerprintTextEnabled = true
+            };
+
+            var contentFingerprint = fingerprinter1.ComputeWeakFingerprint(process, out string fingerprintText);
+            fingerprintText = fingerprintText.ToUpperInvariant();
+
+            var fingerprinter2 = new PipContentFingerprinter(
+                m_context.PathTable,
+                GetDummyContentHash(6),
+                ExtraFingerprintSalts.Default(),
+                pathExpander: expander)
+            {
+                FingerprintTextEnabled = true
+            };
+
+            var contentFingerprint2 = fingerprinter2.ComputeWeakFingerprint(process, out string fingerprintText2);
+            fingerprintText2 = fingerprintText2.ToUpperInvariant();
+
+            XAssert.AreEqual(contentFingerprint, contentFingerprint2);
+            XAssert.AreEqual(fingerprintText, fingerprintText2);
+        }
+
+        [Fact]
+        public void TestModifiedExecutableInsideUnrackedScopeStaysOutOfFingerprints()

```

Comment:
as suggested earlier, let's not copy-paste test methods and instead use a single `Theory` method